packages/ast/test/traverse.js

```diff
@@ -47,6 +47,13 @@ describe("AST traverse", () => {
     assert.isTrue(called, "Module visitor has not been called");
   });
 
+  it("should throw if an unknown node type is encountered", () => {
+    const node = t.stringLiteral("fish");
+    node.type = "foo";
+
+    assert.throws(() => traverse(node, {}), "Unexpected node type foo");
```

Comment:
My initial comment was about:
```js
traverse(a, {
  NotANode() {}
})
```

Where we can check if `type` is in `unionTypesMap`.

Reply:
But it's nice to assert for internal errors too

Reply:
Ah, I was looking at the problem from the other direction! I've added that check too - although it did require a bit more structure:

https://github.com/xtuc/webassemblyjs/pull/349/commits/7d159dd1d9578478e26b3503be1f89cc2f4035ce#diff-a8223d97b02cf2fd47dbcdb8869579c8R997